crd/nodenetworkconfig/client.go

```diff
@@ -54,18 +86,53 @@ func (c *Client) InstallOrUpdate(ctx context.Context) (*v1.CustomResourceDefinit
 		return current, err
 	}
 	if current == nil {
-		current, err = c.crd.Get(ctx, nnc.Name, metav1.GetOptions{})
+		current, err = c.crdcli.Get(ctx, nnc.Name, metav1.GetOptions{})
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to get existing nnc crd")
 		}
 	}
 	if !reflect.DeepEqual(nnc.Spec.Versions, current.Spec.Versions) {
 		nnc.SetResourceVersion(current.GetResourceVersion())
 		previous := *current
-		current, err = c.crd.Update(ctx, nnc, metav1.UpdateOptions{})
+		current, err = c.crdcli.Update(ctx, nnc, metav1.UpdateOptions{})
 		if err != nil {
 			return &previous, errors.Wrap(err, "failed to update existing nnc crd")
 		}
 	}
 	return current, nil
 }
+
+// PatchSpec performs a server-side patch of the passed NodeNetworkConfigSpec to the NodeNetworkConfig specified by the NamespacedName.
+func (c *Client) PatchSpec(ctx context.Context, key types.NamespacedName, spec *v1alpha.NodeNetworkConfigSpec) (*v1alpha.NodeNetworkConfig, error) {
+	obj := &v1alpha.NodeNetworkConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      key.Name,
+			Namespace: key.Namespace,
+		},
+	}
+
+	patch, err := specToJSON(spec)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := c.nnccli.Patch(ctx, obj, ctrlcli.RawPatch(types.ApplyPatchType, patch)); err != nil {
+		return nil, errors.Wrap(err, "failed to patch nnc")
+	}
+
+	return obj, nil
+}
+
+// UpdateSpec does a fetch, deepcopy, and update of the NodeNetworkConfig with the passed spec.
```

Comment:
Update does a resourceVersion check that Patch does not do, right? In which case we may not want to deprecate?

Reply:
the problem is that this update process operates client-side. patch can be server-side. so we don't run in to race/stale-data issues